wp/lib/bap_wp/src/precondition.ml

```diff
@@ -1310,28 +1310,33 @@ let init_mem_range (ctx : Z3.context) (init_mem : bool)
     (* The mem read does not take place in the initialized section. *)
     Bool.mk_false ctx
 
-let init_mem (env : Env.t) (mem : value memmap) : Constr.t list * Env.t =
+let init_mem ?(code_addrs : Utils.Code_addrs.t = Utils.Code_addrs.empty)
```

Comment:
Do we ever anticipate `code_addrs` to be empty in practice? Or is it just for the test suite?

Reply:
Mainly for the test suite. It currently gets computed and cached no matter what, but an idea would be to not do this if the `init-mem` parameter is disabled.

Reply:
I think `Pre.init_mem` doesn't get called when the `init_mem` parameter is disabled. In the test suite, I noticed that `Code_addrs.empty` is provided. Is it necessary to keep this argument optional?

Reply:
That's a fair point. How does the cache behave if the code addrs aren't initially included in the cache and the user runs WP again with the `init_mem` flag on?